core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -85,4 +85,7 @@ private TableProperties() {}
 
   public static final String MANIFEST_LISTS_ENABLED = "write.manifest-lists.enabled";
   public static final boolean MANIFEST_LISTS_ENABLED_DEFAULT = true;
+
+  public static final String METADATA_COMPRESSION_ENABLED = "metadata.compression.enabled";
```

Comment:
What about using `write.metadata.compression-codec` to match the settings for Parquet and Avro?

Reply:
What about `write.metadata.compression.codec` in case we will have more properties related to compression? Are you proposing to make this a string property instead of boolean?

Reply:
Do we want to have an enum of possible values?

Reply:
Let's try to use the same format as other properties. Changing `-` to `.` in this case would be a little confusing. For now, let's just support `none` (default) and `gzip`. We can add more later.

Also, we can add this to the [docs](https://github.com/apache/incubator-iceberg/blob/master/site/docs/configuration.md)!